examples/ConfigurationStoreBackup/BackupAppConfigurationStore.cs

```diff
@@ -0,0 +1,179 @@
+/*
+ * 
+ * 
+	1. We set up Event Grid events emitted by Azure App Configuration to be routed to an Azure Storage queue. Note that event order is not guaranteed by Event grid or storage queue.  
+	2. We create a timer triggered Azure function to read contents of storage queue every <10> minutes. In this tutorial, we are using the same storage account for this Azure function and Storage Queue. 
+	3. Azure function checks if there are any messages from event grid in the queue. Any other messages in the queue in invalid format will be discarded.
+	4. If there are any messages in the queue, retrieve a batch of maximum 32 messages at a time for processing.
+	5. For all retrieved messages, extract the key+label from event message and store them in a hash set. We need to keep track of any unique key+label that was modified or deleted.
+	6. Fetch all settings from primary store. 
+	7. Update only those settings in secondary store which have a corresponding event in the storage queue.
+	8. Delete all settings that were present in storage queue but not in primary store.
+    9. Delete all messages that we received in this batch from the storage queue.
+    10. Repeat from step 4 if there are any messages remaining in storage queue.
+ *
+ *
+ */
+
+using Azure;
+using Azure.Data.AppConfiguration;
+using Azure.Identity;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConfigurationStoreBackup
+{
+    public static class BackupAppConfigurationStore
+    {
+        private static readonly ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("PrimaryStoreEndpoint")), new ManagedIdentityCredential());
+        private static readonly ConfigurationClient secondaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("SecondaryStoreEndpoint")), new ManagedIdentityCredential());
+        private static readonly QueueClient queueClient = new QueueClient(Environment.GetEnvironmentVariable("AzureWebJobsStorage"), "eventgridqueue");
+        private const int maxMessagesToRead = 32;
+
+        [FunctionName("BackupAppConfigurationStore")]
+        public static void Run([TimerTrigger("0 */10 * * * *")]TimerInfo myTimer, ILogger log)
```

Comment:
I'm wondering whether it will work if we change this to async. Can you give it a try since you are on it?
```cs
public static async Task Run(...)
```

Reply:
Yes, this should work. Azure functions could be marked async.

Reply:
I wonder though if the function was made synchronous on purpose since I see this comment in this method implementation.

`Block the thread so that another function instance cannot be triggered while this is still processing.`

Reply:
Yes, we can make the function async, but we want only one function instance to process events at a time, which is the intended behavior of timer triggered functions.

Reply:
Let's say a backup operation takes 30s and this Function is triggered every 10s. Even in the sync version of the Function, what happens? Will the Function not be triggered until the previous one is completed? 

In the same scenario but an async version of the Function, we will `await` on the `BackupAppConfigurationStoreAsync`. Will the Function be triggered with overlaps?

Reply:
By design, only one instance of a timer-triggered Azure Function can be active at a time. For the sync version, the subsequent instances of a function will be skipped if a previous instance is running. I would expect the same behavior for async version too, since the function would still be technically running when the next instance is triggered, but we might want to confirm it.

Reply:
Confirmed using a test function that async version of this function will also be triggered only one instance at a time.

Reply:
Great. Thanks for the confirmation, @avanigupta. In this case, we can have a more natural async flow code.